src/components/HeaderBreadcrumbs/HeaderBreadcrumbs.tsx

```diff
@@ -10,28 +12,52 @@ import './HeaderBreadcrumbs.scss';
 
 const b = block('header-breadcrumbs');
 
+const isCurrentPage = (url: string): boolean => {
+    if (url === '' || url === '.' || url.startsWith('#')) {
+        return true;
+    }
+
+    if (typeof window === 'undefined' || url.startsWith('.')) {
+        return false;
+    }
+
+    const location = window.location;
```

Comment:
use window only in useEffect please

In hook you need to set current page url state and next, in your function you need to compare current url with item.url

Reply:
we are in need of doing it this way specifically, because i cannot think of a simple and guaranteed way of subscribing to `window.location` changes. thus if there is a transition between pages in an SPA, we have no way of updating our current location with it.